lib/datadog/core/configuration/settings.rb

```diff
@@ -322,32 +338,32 @@ def initialize(*_)
 
             # Can be used to enable/disable the Datadog::Profiling.allocation_count feature.
             #
-            # This feature is now controlled via {:experimental_allocation_enabled}
+            # @deprecated Use {:allocation_enabled} (outside of advanced section) instead.
             option :allocation_counting_enabled do |o|
               o.after_set do |_, _, precedence|
                 unless precedence == Datadog::Core::Configuration::Option::Precedence::DEFAULT
                   Datadog.logger.warn(
                     'The profiling.advanced.allocation_counting_enabled setting has been deprecated for removal and no ' \
                     'longer does anything. Please remove it from your Datadog.configure block. ' \
-                    'Allocation counting is now controlled by the `experimental_allocation_enabled` setting instead.'
+                    'Allocation counting is now controlled by the profiling.allocation_enabled setting instead.'
                   )
                 end
               end
             end
 
-            # Can be used to enable/disable collection of allocation profiles.
-            #
-            # This feature is alpha and disabled by default
-            #
-            # @warn This feature is not supported/safe in all Rubies. Details in {Datadog::Profiling::Component} but
-            #       in summary, this should be supported on Ruby 2.x, 3.1.4+, 3.2.3+ and 3.3.0+. Enabling it on
-            #       unsupported Rubies may result in unexpected behaviour, including crashes.
-            #
-            # @default `DD_PROFILING_EXPERIMENTAL_ALLOCATION_ENABLED` environment variable as a boolean, otherwise `false`
+            # @deprecated Use {:allocation_enabled} (outside of advanced section) instead.
             option :experimental_allocation_enabled do |o|
               o.type :bool
-              o.env 'DD_PROFILING_EXPERIMENTAL_ALLOCATION_ENABLED'
               o.default false
+              o.after_set do |_, _, precedence|
+                unless precedence == Datadog::Core::Configuration::Option::Precedence::DEFAULT
+                  Datadog.logger.warn(
+                    'The profiling.advanced.experimental_allocation_enabled setting has been deprecated for removal and ' \
+                    'no longer does anything. Please remove it from your Datadog.configure block. ' \
+                    'Allocation profiling is now controlled by the profiling.allocation_enabled setting instead.'
+                  )
```

Comment:
Related to your PR description, is the problem that `set_option(:allocation_enabled, value)` would act on the `advanced` settings, not on the parent `profiling` settings?

I'm still not fully acquainted with the option code but it could be interesting if you could do `allocation_enabled_option = option :allocation_enabled` and then call `allocation_enabled_option.set(value)` from here. Although yeah if not heavily restricted I can see how this can create a mess of setting value discoverability over time 😂

Reply:
> Related to your PR description, is the problem that set_option(:allocation_enabled, value) would act on the advanced settings, not on the parent profiling settings?

Yes, exactly -- unless I saw it completely wrong, the option groups objects form a tree and are linked only to their children.

> I'm still not fully acquainted with the option code but it could be interesting if you could do allocation_enabled_option = option :allocation_enabled and then call allocation_enabled_option.set(value) from here. Although yeah if not heavily restricted I can see how this can create a mess of setting value discoverability over time 😂

Hmm... I'll admit I didn't try something like this. Let me give a final try to see if I can make it work.

Reply:
Ok so after thinking about it some more, here's how far I got:

1. `allocation_enabled_option = option :allocation_enabled` was a good try, but doesn't quite work because we get an `OptionDefinition` instance. The actual `Option` instance that keeps the data is created separately (in `resolve_option`) and so we don't actually have the object we need at this point.

2. It's possible to add a link to the parent group when defining the options -- [here](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/core/configuration/base.rb#L41) but I'm somewhat reluctant to add a feature to the options system (which is already quite complex) just for this

3. One thing that occurred to me would be inverting the order of the options. That is, `allocation_enabled`, since it's higher in the tree, is able to access the `experimental_allocation_enabled`, so one option would be for `allocation_enabled` to write its result into `experimental_allocation_enabled` and then have the code read the result from `experimental_allocation_enabled` until we deprecated it. That... would somewhat work, although it would interfere with the deprecation warning messages (e.g. not sure if we could still have them).

Overall... Given the current number of users we have using the experimental feature and that we're going to remove the option soon (e.g. for 2.0), I'm not sure it's worth trying really hard to support reading from the old config. (But let me know if you disagree!)

And overall our settings system is too damn complex >_>

Reply:
Given you approved, I'm going to go ahead and merge this, but happy to keep iterating on it as a separate PR if you're not convinced by my attempts above ;)